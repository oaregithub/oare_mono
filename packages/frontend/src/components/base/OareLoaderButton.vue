<template>
  <v-btn v-bind="$attrs" :disabled="loading" @click="$emit('click', $event)">
    <v-progress-circular
      v-if="loading"
      indeterminate
      size="25"
      width="3"
      class="test-spinner"
    />
    <slot v-else></slot>
  </v-btn>
</template>

<script lang="ts">
import { defineComponent } from "@vue/composition-api";

export default defineComponent({
  name: "OareLoaderButton",
  props: {
    loading: {
      type: Boolean,
      default: false,
    },
  },
});
</script>
