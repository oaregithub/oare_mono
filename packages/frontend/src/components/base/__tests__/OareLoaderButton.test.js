import Vuetify from "vuetify";
import VueCompositionApi from "@vue/composition-api";
import { mount, createLocalVue } from "@vue/test-utils";
import OareLoaderButton from "@/components/base/OareLoaderButton";

const vuetify = new Vuetify();
const localVue = createLocalVue();
localVue.use(VueCompositionApi);

describe("OareLoaderButton test", () => {
  const createWrapper = (loading) =>
    mount(OareLoaderButton, {
      localVue,
      vuetify,
      propsData: {
        loading,
      },
      slots: {
        default: "Test Button",
      },
    });

  it("matches snapshot", () => {
    expect(createWrapper(false)).toMatchSnapshot();
  });

  it("displays given text when not loading", () => {
    const wrapper = createWrapper(false);
    expect(wrapper.text()).toBe("Test Button");
  });

  it("displays circular progress when loading", () => {
    const wrapper = createWrapper(true);
    expect(wrapper.find(".test-spinner").exists()).toBe(true);
  });
});
