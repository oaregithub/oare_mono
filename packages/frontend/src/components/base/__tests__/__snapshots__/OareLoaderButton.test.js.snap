// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OareLoaderButton test matches snapshot 1`] = `<button type="button" class="v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"><span class="v-btn__content">Test Button</span></button>`;
